x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/MlConfigVersion.java

```diff
@@ -289,7 +289,7 @@ public static Tuple<MlConfigVersion, MlConfigVersion> getMinMaxMlConfigVersion(D
                 if (mlConfigVersion.after(maxMlConfigVersion)) {
                     maxMlConfigVersion = mlConfigVersion;
                 }
-            } catch (IllegalArgumentException e) {
+            } catch (IllegalStateException e) {
```

Comment:
In this specific bug case this exception will bubble up to the `IngestService` which is running as a `ClusterStateApplier`. We really should not be throwing exceptions on the cluster state application thread unless we are ok with the node being permanently ostracized from the cluster. Can we add some additional logic to the inference processor to make sure that if this exception ever is raised for whatever reason the node is capable of continuing on in the cluster?

Reply:
> this exception will bubble up to the IngestService which is running as a ClusterStateApplier.

I'm not sure I understand this sentence correctly.
The intention of this PR is that the `IllegalStateException` (which is thrown in line 304 below) **does not** bubble up but is caught right here.

> We really should not be throwing exceptions on the cluster state application thread unless we are ok with the node being permanently ostracized from the cluster.

+1

> Can we add some additional logic to the inference processor to make sure that if this exception ever is raised for whatever reason the node is capable of continuing on in the cluster?

I could move the `getMinMlConfigVersion` call under "try-catch" if that's what you mean.
Currently it is:
```
            minNodeVersion = MlConfigVersion.getMinMlConfigVersion(state.nodes());
            try {
                currentInferenceProcessors = InferenceProcessorInfoExtractor.countInferenceProcessors(state);
            } catch (Exception ex) {
                // We cannot throw any exception here. It might break other pipelines.
                logger.debug("failed gathering processors for pipelines", ex);
            }
```

but I could change it to:

```
            try {
              minNodeVersion = MlConfigVersion.getMinMlConfigVersion(state.nodes());
                currentInferenceProcessors = InferenceProcessorInfoExtractor.countInferenceProcessors(state);
            } catch (Exception ex) {
                // We cannot throw any exception here. It might break other pipelines.
                logger.debug("failed gathering processors for pipelines", ex);
            }
```

Reply:
FYI: https://github.com/elastic/elasticsearch/pull/105391